api/src/utils/apply-query.ts

```diff
@@ -765,14 +765,30 @@ export async function applySearch(
 				this.orWhereRaw(`LOWER(??) LIKE ?`, [`${collection}.${name}`, `%${searchQuery.toLowerCase()}%`]);
 			} else if (['bigInteger', 'integer', 'decimal', 'float'].includes(field.type)) {
 				const number = Number(searchQuery);
-				if (!isNaN(number)) this.orWhere({ [`${collection}.${name}`]: number });
+
+				// only cast finite base10 numeric values
+				if (validateNumber(searchQuery, number)) {
+					this.orWhere({ [`${collection}.${name}`]: number });
+				}
 			} else if (field.type === 'uuid' && validate(searchQuery)) {
 				this.orWhere({ [`${collection}.${name}`]: searchQuery });
 			}
 		});
 	});
 }
 
+function validateNumber(value: string, parsed: number) {
+	if (isNaN(parsed) || !Number.isFinite(parsed)) return false;
+
+	// compare without the decimal point
+	if (value.includes('.')) {
+		value = value.replace('.', '');
+		parsed = Number(value);
+	}
+
```

Comment:
Could we add a comment like this?
```suggestion

	// casting parsed value back to string should be equal the original value
	// (prevent unintended number parsing)
```

Reply:
intended for the return statement below what github shows i take it? Then yes that would be a nice addition. Maybe even add a little mention of a hex value or example for clarity 🤔